src/teacher.rb

```diff
@@ -2,7 +2,7 @@
 
 class Teacher < Person
   def initialize(age, specialization, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
+    super(name, age, parent_permission: parent_permission)
     @specialization = specialization
   end
```

Comment:
- [ ] Kindly re-order your parameter here as well. The name is being assigned to age and vice versa.

  ```
  super(age, name, parent_permission: parent_permission)
  ```